src/Integration/UserSettingsConfiguration/SolutionSettingsStorage.cs

```diff
@@ -0,0 +1,160 @@
+
+[Export(typeof(ISolutionSettingsStorage))]
+[PartCreationPolicy(CreationPolicy.Shared)]
+internal sealed class SolutionSettingsStorage : ISolutionSettingsStorage
+{
+    private const string SolutionSettingsFolderName = "SolutionSettings";
+    private const string SettingsFileName = "settings.json";
+    private readonly IActiveSolutionTracker activeSolutionTracker;
+    private readonly ISingleFileMonitorFactory fileMonitorFactory;
+    private readonly IFileSystem fileSystem;
+    private readonly IAnalysisSettingsSerializer serializer;
+
+    private string appDataRoot;
+    private bool disposed;
+    private (string settingsFile, string generatedConfigsBaseDirectory)? solutionFilePaths;
+    private ISingleFileMonitor fileMonitor;
+
+    [ImportingConstructor]
+    public SolutionSettingsStorage(
+        IActiveSolutionTracker activeSolutionTracker,
+        ISingleFileMonitorFactory singleFileMonitorFactory,
+        IFileSystemService fileSystem,
+        IEnvironmentVariableProvider environmentVariableProvider,
+        IAnalysisSettingsSerializer serializer,
+        IInitializationProcessorFactory processorFactory)
+    {
+        this.activeSolutionTracker = activeSolutionTracker;
+        this.fileSystem = fileSystem;
+        this.serializer = serializer;
+        fileMonitorFactory = singleFileMonitorFactory;
+        InitializationProcessor = processorFactory.CreateAndStart<SolutionSettingsStorage>(
+            [activeSolutionTracker],
+            () =>
+            {
+                if (disposed)
+                {
+                    return;
+                }
+
+                // Note: the data is stored in the roaming profile so it will be sync across machines for domain-joined users.
+                appDataRoot = environmentVariableProvider.GetSLVSAppDataRootPath();
+                CreateSolutionSettingsMonitorAndSubscribe(activeSolutionTracker.CurrentSolutionName);
+                activeSolutionTracker.ActiveSolutionChanged += ActiveSolutionTrackerOnActiveSolutionChanged;
+            });
+    }
+
+    public IInitializationProcessor InitializationProcessor { get; }
+    public string SettingsFilePath => solutionFilePaths?.settingsFile;
+    public string ConfigurationBaseDirectory => solutionFilePaths?.generatedConfigsBaseDirectory;
+    public event EventHandler SettingsFileChanged;
+
+    public void EnsureSettingsFileExists()
+    {
+        if (SettingsFilePath == null)
+        {
+            return;
+        }
+
+        if (!fileSystem.File.Exists(SettingsFilePath))
+        {
+            serializer.SafeSave(SettingsFilePath, new SolutionAnalysisSettings());
+        }
+    }
+
+    public void SaveSettingsFile(SolutionAnalysisSettings settings)
+    {
+        serializer.SafeSave(SettingsFilePath, settings);
+        InvokeSettingsFileChanged();
+    }
+
+    public SolutionAnalysisSettings LoadSettingsFile() => serializer.SafeLoad<SolutionAnalysisSettings>(SettingsFilePath);
```

Comment:
If this value is cached here, AnalysisSettings.SolutionFileExclusions can be removed and SolutionSettingsProvider can use this value directly instead of going through UserSettingsProvider